flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureUtils.java

```diff
@@ -969,4 +970,29 @@ public void run() {
 			return DELAYER.schedule(runnable, delay, timeUnit);
 		}
 	}
+
+	/**
+	 * Asserts that the given {@link CompletableFuture} is not completed exceptionally. If the future
+	 * is completed exceptionally, then it will call the {@link FatalExitExceptionHandler}.
+	 *
+	 * @param completableFuture to assert for no exceptions
+	 */
+	public static void assertNoException(CompletableFuture<?> completableFuture) {
+		assertNoException(completableFuture, FatalExitExceptionHandler.INSTANCE);
+	}
+
+	/**
+	 * Asserts that the given {@link CompletableFuture} is not completed exceptionally. If the future
+	 * is completed exceptionally, then it will call the given uncaught exception handler.
+	 *
+	 * @param completableFuture to assert for no exceptions
+	 * @param uncaughtExceptionHandler to call if the future is completed exceptionally
+	 */
+	public static void assertNoException(CompletableFuture<?> completableFuture, Thread.UncaughtExceptionHandler uncaughtExceptionHandler) {
```

Comment:
The name implies a `boolean` return type.

Reply:
I could rename the method to `checkNoException`. Would that be better?

Reply:
I think what bothers me is that this overload might not assert or check anything depending on the implementation of the handler. If this overload is only for testing purposes, a `@VisibleForTesting` would suffice.
The other overload, however, exits the JVM, which can be interpreted as an assertion/check.

Reply:
Gotcha. I think you are right. What about calling this variant `handleUncaughtException`?

Reply:
ok